crypto/dh/dh_key.c

```diff
@@ -60,6 +67,13 @@ int ossl_dh_compute_key(unsigned char *key, const BIGNUM *pub_key, DH *dh)
         return 0;
     }
 
+#ifdef FIPS_MODULE
+    if (DH_check_pub_key(dh, pub_key, &validate) <= 0) {
+        ERR_raise(ERR_LIB_DH, DH_R_CHECK_PUBKEY_INVALID);
+        return 0;
+    }
+#endif
+
```

Comment:
This is a very expensive operation which is really unnecessary for the known safe prime DH parameters and furthermore you do not check the validate value so you do not really reject bad pub keys at all.

You should instead use ossl_ffc_validate_public_key_partial() as the only FIPS allowed DH parameters are the known safe primes now (aren't they?).